tsdb/wlog/watcher.go

```diff
@@ -420,9 +437,31 @@ func (w *Watcher) watch(segmentNum int, tail bool) error {
 
 			return nil
 
+		// we haven't read due to a notification in quite some time, try reading anyways
 		case <-readTicker.C:
```

Comment:
You could do `case <-readTicker.C, <-w.ReadNotify:` here, but then you wouldn't get to log which event caused the read (if you want to keep that).